src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2504,6 +2504,28 @@ uint32 Creature::GetPetAutoSpellOnPos(uint8 pos) const
         return m_charmInfo->GetCharmSpell(pos)->GetAction();
 }
 
+float Creature::GetPetChaseDistance() const
+{
+    float range = MELEE_RANGE;
+
+    for (uint8 i = 0; i < GetPetAutoSpellSize(); ++i)
+    {
+        uint32 spellID = GetPetAutoSpellOnPos(i);
+        if (!spellID)
+            continue;
+
+        if (SpellInfo const* spellInfo = sSpellMgr->GetSpellInfo(spellID))
+        {
+            if (spellInfo->GetRecoveryTime() == 0 &&  // No cooldown
+                    spellInfo->RangeEntry->ID != 1 /*Self*/ && spellInfo->RangeEntry->ID != 2 /*Combat Range*/ &&  // Range bigger than melee
+                        range < spellInfo->GetMaxRange(true))  // Bigger than previous
+                range = spellInfo->RangeEntry->minRangeHostile;
+        }
+    }
+
+    return range;
```

Comment:
yeah, it depends on which spell is activated as autocast.
Example:
summon water elemental -> autocast activated -> it will keep the spell range while chasing
-> deactivate autocast -> will chase on melee range